cli/azd/pkg/azsdk/zip_deploy_client.go

```
@@ -114,6 +122,202 @@ func (c *ZipDeployClient) BeginDeploy(
+	case armappservice.DeploymentBuildStatusBuildInProgress:
+		progressLog("Running build process")
+		return nil
+	case armappservice.DeploymentBuildStatusRuntimeStarting:
+		progressLog(fmt.Sprintf("Starting runtime process, %d in progress instances, %d successful instances",
+			inProgressNumber, successNumber))
+		return nil
+	case armappservice.DeploymentBuildStatusRuntimeSuccessful, armappservice.DeploymentBuildStatusBuildSuccessful:
+		return nil
+	case armappservice.DeploymentBuildStatusRuntimeFailed:
+		totalNumber := inProgressNumber + successNumber + failNumber
+
+		if successNumber > 0 {
+			errorString += fmt.Sprintf("%d/%d instances failed to start successfully\n", failNumber, totalNumber)
+		} else if totalNumber > 0 {
+			errorString += fmt.Sprintf("Deployment failed because the runtime process failed. In progress instances: %d, "+
+				"Successful instances: %d, Failed Instances: %d\n",
+				inProgressNumber, successNumber, failNumber)
+		}
+
+		logErrorFunction(properties, "runtime ")
+		return fmt.Errorf(errorString)
+	case armappservice.DeploymentBuildStatusBuildFailed:
+		errorString += "Deployment failed because the build process failed\n"
+		logErrorFunction(properties, "build ")
+		return fmt.Errorf(errorString)
+	// Progress Log for other states
+	default:
+		if len(status) > 0 {
+			progressLog(fmt.Sprintf("Running deployment status api in stage %s", status))
+		}
+		return nil
+	}
+}
+
+func (c *ZipDeployClient) DeployTrackStatus(
+	ctx context.Context,
+	zipFile io.ReadSeeker,
+	subscriptionId string,
+	resourceGroup string,
+	appName string,
+	progressLog func(string)) error {
+	var response armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse
+
+	poller, err := c.BeginDeployTrackStatus(ctx, zipFile, subscriptionId, resourceGroup, appName)
+	if err != nil {
+		return err
+	}
+
+	delay := 3 * time.Second
+	pollCount := 0
+	for {
+		var resp *http.Response
+
+		resp, err = poller.Poll(ctx)
+		if err != nil {
+			break
```

I think this may be a problem, as the logic ends up swallowing the `err` returned here, and would return `nil` as a success?